internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -343,6 +353,13 @@ func (r ContainerAppEnvironmentResource) Read() sdk.ResourceFunc {
 				state.ResourceGroup = id.ResourceGroupName
 				state.Location = location.Normalize(model.Location)
 				state.Tags = tags.Flatten(model.Tags)
+				if model.Identity != nil {
+					ident, err := identity.FlattenSystemAndUserAssignedMapToModel(pointer.To(identity.SystemAndUserAssignedMap(*model.Identity)))
+					if err != nil {
+						return err
+					}
+					state.Identity = pointer.From(ident)
```

Comment:
Here we can use the legacy flatten func, could you also wrap the error with `` flattening `identity` ... ``?

```suggestion
					ident, err := identity.FlattenLegacySystemAndUserAssignedMapToModel(model.Identity)
					if err != nil {
						return fmt.Errorf("flattening `identity`: %+v", err)
					}
					state.Identity = ident
```

Reply:
Will fix.